pkg/upgrade/upgrademanager/manager.go

```diff
@@ -237,6 +238,51 @@ func (m *Manager) RunPermanentUpgrades(ctx context.Context, upToVersion roachpb.
 	}
 
 	for _, u := range permanentUpgrades {
+		// Check whether a 22.2 or older node has already run the last
+		// startupmigration that was added before the permanent upgrades framework
+		// was introduced. If it has, we don't want to run this upgrade. Allowing
```

Comment:
> this upgrade

The comment says "this upgrade" but we mean _all_ the permanent upgrades, right? since `startupMigrationAlreadyRan` being true in one means it'll be true in any right?

Reply:
Thanks!

I think it'd be good to spell out that none of the permeant upgrades are actually run if we detect the legacy key (and why that is correct, i.e. that all the permanent upgrade in 24.1 are also captured in the legacy upgrades?)